### PR TITLE
Persistence stale rowsets meta

### LINKED ISSUE #4453: Persistence stale rowsets meta
**Description**
Persistence stale rowsets meta. When be reboots, stale rowsets meta can resume and the stale version can also be readable before stale gc time.

### PR DESCRIPTION
**Pull request by ZhangYu0123**
## Proposed changes

Persistence stale rowsets meta. When be reboots, stale rowsets meta can resume and the stale version can also be readable before stale gc time.  

ISSUE: #4453

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #4453), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

**Comment by kangkaisen** [comment]
@ZhangYu0123 Hi, What's the concrete issue that this PR want to Fix? what's the accurate definition for `stale rowsets` ?

**Comment by ZhangYu0123** [comment]
> @ZhangYu0123 Hi, What's the concrete issue that this PR want to Fix? what's the accurate definition for `stale rowsets` ?

https://github.com/apache/incubator-doris/issues/4017

**Comment by kangkaisen** [comment]
> > @ZhangYu0123 Hi, What's the concrete issue that this PR want to Fix? what's the accurate definition for `stale rowsets` ?
> 
> #4017

I see. Thanks. 
But I think the `stale rowsets` naming isn't very appropriate. semantically, stale means the data isn't right and we shouldn't read it.

File: be/src/olap/tablet_meta.h [hunk 1]
```diff
@@ -171,8 +171,9 @@ class TabletMeta {
                          const std::vector<RowsetMetaSharedPtr>& to_delete);
     void revise_rs_metas(std::vector<RowsetMetaSharedPtr>&& rs_metas);
 
-
     void revise_inc_rs_metas(std::vector<RowsetMetaSharedPtr>&& rs_metas);
+    void revise_stale_rs_metas(std::vector<RowsetMetaSharedPtr>&& rs_metas);
```
Comment (morningman) [inline be/src/olap/tablet_meta.h:175]: This function seems not use?

Comment (morningman) [inline be/src/olap/tablet_meta.h:175]: This function seems not use?

File: be/src/olap/tablet.cpp [hunk 2]
```diff
@@ -108,6 +109,22 @@ OLAPStatus Tablet::_init_once_action() {
         _inc_rs_version_map[version] = std::move(rowset);
     }
 
+    // init stale rowset
+    for (auto& stale_rs_meta : _tablet_meta->all_stale_rs_metas()) {
+        Version version = stale_rs_meta->version();
+        RowsetSharedPtr rowset = get_stale_rowset_by_version(version);
+        if (rowset == nullptr) {
```
Comment (morningman) [inline be/src/olap/tablet.cpp:116]: Under what circumstances will it not be found?

Comment (morningman) [inline be/src/olap/tablet.cpp:116]: Under what circumstances will it not be found?

File: be/src/olap/version_graph.cpp [hunk 3]
```diff
@@ -43,6 +43,167 @@ void TimestampedVersionTracker::construct_versioned_tracker(const std::vector<Ro
...
+        } else {
+            std::unordered_map<int64_t, RowsetMetaSharedPtr> item;
+            item[stale_meta->end_version()] = stale_meta;
+            stale_map[stale_meta->start_version()] = std::move(item);
+        }
+        // 2.3 add version to version_graph
+        Version stale_meta_version = stale_meta->version();
+        add_version(stale_meta_version);
+        // 2.4 find the path
+        if (r) {
```
Comment (morningman) [inline be/src/olap/version_graph.cpp:117]: I didn't get this logic...

Comment (morningman) [inline be/src/olap/version_graph.cpp:117]: I didn't get this logic...

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
> > > @ZhangYu0123 Hi, What's the concrete issue that this PR want to Fix? what's the accurate definition for `stale rowsets` ?
> > 
> > 
> > #4017
> 
> I see. Thanks.
> But I think the `stale rowsets` naming isn't very appropriate. semantically, stale means the data isn't right and we shouldn't read it.

Hi @kangkaisen , I think `stale` means `not fresh`, but not means `incorrect`, so it looks good to me.

**Pull request merged**